pandas/core/generic.py

```diff
@@ -7922,34 +7922,97 @@ def last(self, offset):
     def rank(self, axis=0, method='average', numeric_only=None,
              na_option='keep', ascending=True, pct=False):
         """
-        Compute numerical data ranks (1 through n) along axis. Equal values are
-        assigned a rank that is the average of the ranks of those values
+        Compute numerical data ranks (1 through n) along axis.
+
+        By default, equal values are assigned a rank that is the average of the
+        ranks of those values.
 
         Parameters
         ----------
         axis : {0 or 'index', 1 or 'columns'}, default 0
-            index to direct ranking
-        method : {'average', 'min', 'max', 'first', 'dense'}
+            Index to direct ranking.
+        method : {'average', 'min', 'max', 'first', 'dense'}, default 'average'
             * average: average rank of group
             * min: lowest rank in group
             * max: highest rank in group
             * first: ranks assigned in order they appear in the array
             * dense: like 'min', but rank always increases by 1 between groups
-        numeric_only : boolean, default None
-            Include only float, int, boolean data. Valid only for DataFrame or
-            Panel objects
-        na_option : {'keep', 'top', 'bottom'}
-            * keep: leave NA values where they are
-            * top: smallest rank if ascending
-            * bottom: smallest rank if descending
-        ascending : boolean, default True
-            False for ranks by high (1) to low (N)
-        pct : boolean, default False
-            Computes percentage rank of data
+        numeric_only : bool, optional
+            For DataFrame objects, rank only numeric columns if set to True.
+        na_option : {'keep', 'top', 'bottom'}, default 'keep'
+            How to rank NaN values:
+
+            * keep: assign NaN rank to NaN values
+            * top: assign smallest rank to NaN values if ascending
+            * bottom: assign highest rank to NaN values if ascending
+        ascending : bool, default True
+            False for ranks by high (1) to low (N).
+        pct : bool, default False
+            Computes percentage rank of data.
 
         Returns
         -------
         ranks : same type as caller
+            Return a Series or DataFrame with data ranks as values
+
+        See Also
+        --------
+        pandas.core.groupby.GroupBy.rank : Rank of values within each group.
+
+        Examples
+        --------
+        The default behavior returns average ranks of every column
+
+        >>> df = pd.DataFrame(data={'Customer': [1, 2, 3, 4, 5],
+        ...                         'Tot_Spend': [12, 20, 20, 18, 16]})
```

Comment:
I think having an arbitrary example doesn't make easier to understand the example.

We have in some places data like animals and their number of legs. I think if you see:
```
cat: 2.5
penguin: 4
dog: 2.5
spider: 1
```

Is quite easy to understand that the spider is 1 because it has more legs, and that cat and dog have the same number because they have the same number of legs.

Seeing in the example the client 2 and 3 have the same number, forces me to take a detailed look at the data to understand what is going on.